### PR TITLE
Easify managing containers

### PR DESCRIPTION
**Pull request by jkarolczak**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -9,21 +9,26 @@
...
+        - [nvidia docker runtime](https://developer.nvidia.com/nvidia-container-runtime)
+2. Clone this repository.
+3. `sudo chmod 744 ./start.sh ./stop.sh`
+4. Set the environment configuration in the `docker/.env`:
+    - `PYTORCH`, `CUDA`, `CUDNN` - only for GPU, modify only if needed. Before changing verify against Minkowski Engine
+      requirements
+    - `DATA_PATH` - if your name is Adaś and you are computer you should leave it as it is. If your name is Ania, Witek
+      or Konrad, RNG guesses it is `../../data/`. In other cases set the path to the folder containing data
+5. Start container using `./start.sh`. If you want to run container on cpu use `./start.sh cpu`.
+6. To stop use `./stop.sh` or `./stop.sh` for cpu respectively.
```
Comment (wtaisner) [inline README.md:31]: I think thay You meant `./stop.sh cpu` here

File: .gitignore [hunk 2]
```diff
@@ -32,3 +32,7 @@ blob_details/output/volume_graph.html
 
 # secret configs
 cfg/neptune.yaml
+
+# docker configs
+!docker/compose-cpu.yml
+!docker/compose-gpu.yml
```
Comment (wtaisner) [inline .gitignore:38]: shouldn't we also exclude `docker/.env`? I mean, it has to be changed for each case, yet makes no point in commiting it, perhaps exclude `.env` and add a template?

File: docker/DockerfileCPU [hunk 3]
```diff
@@ -16,9 +16,8 @@ RUN pip install sklearn pandas torchmetrics addict yapf neptune-client h5py
 
 ENV MAX_JOBS=4
 RUN pip install torch torchvision torchaudio --extra-index-url https://download.pytorch.org/whl/cpu
-RUN pip install -U git+https://github.com/NVIDIA/MinkowskiEngine -v --no-deps \
-                           --install-option="--cpu_only" \
-                           --install-option="--blas=openblas"
+RUN git clone --recursive "https://github.com/NVIDIA/MinkowskiEngine"
+RUN cd MinkowskiEngine; python setup.py install --force_cuda --blas=openblas
```
Comment (wtaisner) [inline docker/DockerfileCPU:20]: ain't sure right now, but `--force_cuda` seems to force cuda, but its CPU dockerfile

Comment (wtaisner) [reply docker/DockerfileCPU:20]: at least in my case it fails due to the lack of CUDA_HOME

File: start.sh [hunk 4]
```diff
@@ -0,0 +1,5 @@
+if [[ $1 == "cpu" ]]; then
+    docker compose --env-file docker/.env -f docker/compose-cpu.yml up -d
```
Comment (wtaisner) [inline start.sh:2]: at least with my (default) config, it requires `docker-compose` (with a hyphen)

perhaps @annprzy or @konradszewczyk  can verify, and GPU version along

Comment (wtaisner) [reply start.sh:2]: in my case, the script has a problem with finding files in `docker` directory. I had to run the command manually from inside the `docker` directory, 

https://stackoverflow.com/questions/64499052/docker-compose-env-file-is-not-working-at-all

the solution I found here

File: stop.sh [hunk 5]
```diff
@@ -0,0 +1,5 @@
+if [[ $1 == "cpu" ]]; then
+    docker compose --env-file docker/.env -f docker/compose-cpu.yml down
```
Comment (wtaisner) [inline stop.sh:2]: most probably the same problems as in start.sh, 

i'm once again asking @annprzy or @konradszewczyk for verification

File: docker/DockerfileCPU (hunk 3 above)
Comment (wtaisner) [reply docker/DockerfileCPU:20]: changing to --cpu_only did the trick

Comment (wtaisner) [reply docker/DockerfileCPU:20]: and also, there should be  `python3` instead of `python`

**Review by wtaisner — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (wtaisner) [reply README.md:31]: line 31st should only be mentioned here, specifically second `./stop.sh`

File: start.sh (hunk 4 above)
Comment (jkarolczak) [reply start.sh:2]: The lack of the hyphen is not a mistake. Please see the "important box" here: https://docs.docker.com/compose/reference/

`docker-compose` is used for docker compose v1, which we really don't like

File: docker/DockerfileCPU (hunk 3 above)
Comment (jkarolczak) [reply docker/DockerfileCPU:20]: Mea culpa. Fix will be provided within 2.74 min

File: README.md (hunk 1 above)
Comment (jkarolczak) [reply README.md:31]: Indeed

File: .gitignore (hunk 2 above)
Comment (jkarolczak) [reply .gitignore:38]: Can do.

File: start.sh (hunk 4 above)
Comment (jkarolczak) [reply start.sh:2]: If you need a reference to installation tutorial see: https://docs.docker.com/compose/install/linux/#install-the-plugin-manually

Comment (wtaisner) [reply start.sh:2]: thanks bro, didn't see this

File: cfg/models/minkloc3dv2.yaml [hunk 6]
```diff
@@ -15,4 +15,4 @@ normalize_embeddings: False
 
 input_dim: 1
 
-output_dim: 219
```
[the remark this answers is not in the document]
Comment (wtaisner) [reply cfg/models/minkloc3dv2.yaml:18]: apparently I wanted to mention next line, in which `output_dim: 3` is added

[the remark this answers is not in the document]
Comment (jkarolczak) [reply cfg/models/minkloc3dv2.yaml:18]: You're absolutely right. Done

**Review by wtaisner — APPROVED** [review]

**Comment by jkarolczak** [comment]
Adaś confirms that the new setup works way more than just fine. Greetings from Adaś to the whole world

**Pull request merged**